src/client/pages/Payment/useTrustlyURL.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react'
+import { useLocation } from 'react-router'
+import { useRegisterDirectDebitMutation } from 'data/graphql'
+import { SUCCESS_SUFFIX, FAIL_SUFFIX } from './Payment.constants'
+
+export const useTrustlyURL = () => {
+  const { pathname } = useLocation()
+  const baseURL = `${window.location.origin}${pathname}`
+  const [registerDirectDebit, { data }] = useRegisterDirectDebitMutation({
+    variables: {
+      clientContext: {
+        successUrl: [baseURL, SUCCESS_SUFFIX].join(''),
+        failureUrl: [baseURL, FAIL_SUFFIX].join(''),
+      },
+    },
+  })
+
+  useEffect(() => {
```

Comment:
It non-obvious for caller that useTrustlyUrl mutates something.  Can we call register on caller side?

Reply:
right, I was confused about how to implement this -- it doesn't feel like it fits in a useEffect -- do you have a better idea?

I can also rename this function but I feel like it nice to co-locate it

Reply:
Maybe we can solve this by naming only
```
// Caller side
const trustlyUrl = useResigsterTrustlyPaument()
```
Now it's clear that hook initiates a mutation